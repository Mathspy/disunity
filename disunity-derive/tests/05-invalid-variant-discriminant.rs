use disunity_derive::Variant;

#[derive(Variant)]
enum Class {
    #[disunity(discriminant = "nonsense")]
    GameObject { field: i32 },
}

fn main() {
    let _ = ClassVariant::GameObject;
}
