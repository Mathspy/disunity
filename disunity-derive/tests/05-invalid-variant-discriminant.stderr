error[E0308]: mismatched types
 --> tests/05-invalid-variant-discriminant.rs:5:31
  |
5 |     #[disunity(discriminant = "nonsense")]
  |                               ^^^^^^^^^^ expected `isize`, found `&str`
